src/main/java/org/sonarlint/intellij/trigger/BuildFinishedAnalysisTrigger.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2024 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.trigger;
+
+import com.intellij.compiler.server.BuildManagerListener;
+import com.intellij.openapi.project.Project;
+import java.util.UUID;
+import org.sonarlint.intellij.analysis.AnalysisSubmitter;
+import org.sonarlint.intellij.common.ui.SonarLintConsole;
+import org.sonarlint.intellij.common.util.SonarLintUtils;
+
+public class BuildFinishedAnalysisTrigger implements BuildManagerListener {
+
+  @Override public void buildFinished(Project project, UUID sessionId, boolean isAutomake) {
+    if (!isAutomake) {
+      // covered by CompilationFinishedAnalysisTrigger.compilationFinished
+      return;
+    }
+
+    SonarLintUtils.getService(project, SonarLintConsole.class).debug("build finished");
```

Comment:
is it worth checking for project disposed?

Reply:
I don't know. We were not checking before. Same as my previous comment, let me check first what should be the approach for checking disposed projects